src/lib/components/Button.svelte

```diff
@@ -0,0 +1,46 @@
+<script lang="ts">
+	export let type: 'default' | 'rounded' | 'square' = 'default';
+</script>
+
+<button class="Button {type}" on:click>
+	<slot>Text Label</slot>
+</button>
+
+<style lang="scss">
+	@use './src/lib/styles/colors';
+
+	.Button {
+		color: colors.$white;
+		background-color: colors.$primary;
+		padding: 0.6rem 1rem;
+		border-radius: 10px;
+		border: none;
+		cursor: pointer;
+		transition: 0.15s ease;
+
+		&:disabled {
+			background-color: colors.$disabled;
+			cursor: not-allowed;
+		}
+
+		&:hover {
+			background-color: colors.$hover;
+		}
+
+		&:focus {
```

Comment:
Usually focus default styles are disabled in design specs. We can use `:focus-visible` to retain proper focus styling on keyboard navigation.
See https://developer.mozilla.org/fr/docs/Web/CSS/:focus-visible